app/controllers/incomeSources/add/BusinessAddedObligationsController.scala

```diff
@@ -0,0 +1,156 @@
+
+class BusinessAddedObligationsController @Inject()(authenticate: AuthenticationPredicate,
+                                                   val authorisedFunctions: AuthorisedFunctions,
+                                                   checkSessionTimeout: SessionTimeoutPredicate,
+                                                   retrieveNino: NinoPredicate,
+                                                   val retrieveIncomeSources: IncomeSourceDetailsPredicate,
+                                                   val retrieveBtaNavBar: NavBarPredicate,
+                                                   val itvcErrorHandler: ItvcErrorHandler,
+                                                   incomeSourceDetailsService: IncomeSourceDetailsService,
+                                                   nextUpdatesService: NextUpdatesService,
+                                                   val obligationsView: BusinessAddedObligations)
+                                                  (implicit val appConfig: FrontendAppConfig,
+                                                   implicit val itvcErrorHandlerAgent: AgentItvcErrorHandler,
+                                                   implicit override val mcc: MessagesControllerComponents,
+                                                   val ec: ExecutionContext,
+                                                   dateService: DateServiceInterface)
+  extends ClientConfirmedController with I18nSupport with FeatureSwitching{
+
+  lazy val backUrl = controllers.incomeSources.add.routes.AddBusinessReportingMethod.show().url
+  lazy val agentBackUrl = controllers.incomeSources.add.routes.AddBusinessReportingMethod.showAgent().url
+
+  def show(id: Option[String]): Action[AnyContent] = (checkSessionTimeout andThen authenticate andThen retrieveNino
+    andThen retrieveIncomeSources andThen retrieveBtaNavBar).async {
+    implicit user =>
+      handleRequest(isAgent = false, id)
+  }
+
+  def showAgent(id: Option[String]): Action[AnyContent] =
+    Authenticated.async {
+      implicit request =>
+        implicit user =>
+          getMtdItUserWithIncomeSources(incomeSourceDetailsService) flatMap {
+            implicit mtdItUser =>
+              handleRequest(isAgent = true, id)
+          }
+    }
+
+  def handleRequest(isAgent: Boolean, id: Option[String])(implicit user: MtdItUser[_], ec: ExecutionContext): Future[Result] = {
+    if (isDisabled(IncomeSources)) {
+      Future.successful(Redirect(controllers.routes.HomeController.show()))
+    } else {
+      id match {
+        case None => Logger("application").error(
+          s"[BusinessAddedObligationsController][handleRequest] - Error: No id supplied by reporting method page")
+          Future(itvcErrorHandler.showInternalServerError())
+        case Some(value) => {
+          if (!user.incomeSources.businesses.exists(x => x.incomeSourceId.contains(value))) {
```

Comment:
this is redundant step, as you calling service, and handling this scenario already +
we should make assumption that incomeSource ids under specific user will always belong to the user